protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -173,7 +177,20 @@ public T parse(InputStream stream) {
       try {
         if (stream instanceof KnownLength) {
           int size = stream.available();
-          if (size > 0 && size <= DEFAULT_MAX_MESSAGE_SIZE) {
+          // TODO(chengyuanzhang): we may still want to go with the byte array approach for small
+          //  messages.
+          if (stream instanceof HasByteBuffer
+              && stream.markSupported() && ((HasByteBuffer) stream).getByteBufferSupported()) {
+            List<ByteBuffer> buffers = new ArrayList<>();
+            stream.mark(size);
+            while (stream.available() != 0) {
+              ByteBuffer buffer = ((HasByteBuffer) stream).getByteBuffer();
+              stream.skip(buffer.remaining());
+              buffers.add(buffer);
+            }
+            stream.reset();
```

Comment:
I don't think the reset here really serves any purpose? IIUC the mark is only done to say "don't release the stuff I'm about to read" but it will all be released when the stream is closed once this method returns regardless.

If the reset stays then to be more "correct" shouldn't `stream.skip(size)` also be called after `CodedInputStream.newInstance(buffers)`?

Reply:
Thanks to https://github.com/grpc/grpc-java/pull/7330#issuecomment-675635862.

Reply:
@voidzcy I was referring to this specific line here, not the existence of the `reset()` method. This line could be deleted, right?

Reply:
I see. Deleted.